src/test/java/net/imagej/convert/ConvertROITreeToMaskTest.java

```diff
@@ -0,0 +1,234 @@
+				), //
+				ClosedWritableBox.class //
+			}, //
+			new Object[] { //
+				new OpenWritableBox( //
+					new double[] { 0, 0 }, //
+					new double[] { 5, 10 } //
+				), //
+				OpenWritableBox.class //
+			}, //
+			new Object[] { //
+				new ClosedWritablePolygon2D( //
+					new double[] { 0, 3, 4 }, //
+					new double[] { 0, 0, 4 } //
+				), //
+				ClosedWritablePolygon2D.class //
+			}, //
+			new Object[] { //
+				new OpenWritablePolygon2D( //
+					new double[] { 0, 3, 4 }, //
+					new double[] { 0, 0, 4 } //
+				), //
+				OpenWritablePolygon2D.class //
+			}, //
+			new Object[] { //
+				new DefaultWritableLine( //
+					new double[] { 0, 0 }, //
+					new double[] { 5, 5 }, //
+					true //
+				), //
+				DefaultWritableLine.class, //
+			}, //
+			new Object[] { //
+				new DefaultWritablePolyline(Arrays.asList( //
+					new RealPoint(0, 0), //
+					new RealPoint(1, 1), //
+					new RealPoint(2, 0) //
+				)), //
+				DefaultWritablePolyline.class, //
+			}, //
+			new Object[] { //
+				new RealMask()
+				{
+
+					@Override
+					public boolean test(RealLocalizable realLocalizable) {
+						return realLocalizable.getDoublePosition(0) + realLocalizable
+							.getDoublePosition(1) == 0;
+					}
+
+					@Override
+					public int numDimensions() {
+						return 2;
+					}
+				}, //
+				RealMask.class //
+			}, //
+			new Object[] { //
+				new MaskPredicate<RealLocalizable>()
+				{
+
+					@Override
+					public int numDimensions() {
+						return 2;
+					}
+
+					@Override
+					public boolean test(RealLocalizable realLocalizable) {
+						// true iff any coordinate is zero
+						return Arrays //
+							.stream(realLocalizable.positionAsDoubleArray()) //
+							.anyMatch(d -> d == 0);
+					}
+
+					@Override
+					public MaskPredicate<RealLocalizable> and(
+						Predicate<? super RealLocalizable> other)
+				{
+						throw new UnsupportedOperationException();
+					}
+
+					@Override
+					public MaskPredicate<RealLocalizable> or(
+						Predicate<? super RealLocalizable> other)
+				{
+						throw new UnsupportedOperationException();
+					}
+
+					@Override
+					public MaskPredicate<RealLocalizable> negate() {
+						throw new UnsupportedOperationException();
+					}
+
+					@Override
+					public MaskPredicate<RealLocalizable> minus(
+						Predicate<? super RealLocalizable> other)
+				{
+						throw new UnsupportedOperationException();
+					}
+
+					@Override
+					public MaskPredicate<RealLocalizable> xor(
+						Predicate<? super RealLocalizable> other)
+				{
+						throw new UnsupportedOperationException();
+					}
+				}, //
+				MaskPredicate.class //
+			} //
+		);
+	}
+
+	@Test
+	public void testROITreeToROITest() {
+		// Construct the ROI tree
+		ROITree tree = new DefaultROITree();
+		tree.addROIs(Collections.singletonList(ROI));
+		// Assert that the convertService can convert this tree to a mask
+		final ConvertService convertService = c.service(ConvertService.class);
```

Comment:
How about using `ConvertService#supports()` in addition here?
```suggestion
		// Assert that the convertService can convert this tree to a mask
		final ConvertService convertService = c.service(ConvertService.class);
		Assert.assertTrue(convertService.supports(tree, ROIClass));
```

Reply:
Good idea!

Fixed in e6ed111